### DISCUSSION TITLE
[SOLVED] SDRPP install: Dependency is not satisfied

### DISCUSSION THREAD

**Discussion by DarcSceptor** [AlexandreRouma/SDRPlusPlus#1447] [Q&A] [answered]
I am trying to install SDRPP the Ubuntu Jammy version on Linux Mint 22
I double-click the deb file but I get the error message for dependancy: libvolk2-dev.
I have tried numerous ways to get that installed separately but would prefer you package things to install all dependancies like several other packages I've installed.

So HOW do I install libvolk2-dev? (I'm new to Linux)

**Comment by KubaPro010** [comment]
the dependencies should be automatically installed but when they're not then theyre not, and alexandre came up with a very smart decision to remove the command to install dependencies (the command is `sudo apt install libfftw3-dev libglfw3-dev libvolk2-dev libzstd-dev libairspyhf-dev libiio-dev libad9361-dev librtaudio-dev libhackrf-dev`, and he's gonna try to justify it with a smart sounding reason) 

TL;DR: run `sudo apt install libfftw3-dev libglfw3-dev libvolk2-dev libzstd-dev libairspyhf-dev libiio-dev libad9361-dev librtaudio-dev libhackrf-dev`

**Reply by DarcSceptor** [reply]
Oooops

Package libvolk2-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  libvolk-dev libcpu-features-dev

E: Package 'libvolk2-dev' has no installation candidate
E: Unable to locate package l
ibzstd-dev: command not found

**Reply by KubaPro010** [reply]
try `sudo apt install libfftw3-dev libglfw3-dev libvolk-dev libzstd-dev libairspyhf-dev libiio-dev libad9361-dev librtaudio-dev libhackrf-dev`

**Reply by DarcSceptor** [reply]
Okay...one step down and one more being a pain in the ass.
I am in the terminal. I changed directories to Downloads. I run ls or dir and I get the file for SDRPP ...
If I do sudo apt install sdrpp_ubuntu_jammy_amd64.deb it says unable to locate package
If I double click the deb file it comes up and says that I am still missing libvolk2-dev.

**Reply by DarcSceptor** [reply, marked as answer]
I finally got the package to install. Here is what is needed as it seems there are important packages missed by the Mint dev team:

`sudo apt-get install libpango1.0-dev libcairo2-dev`

**Reply by Hue-Jhan** [reply]
> I finally got the package to install. Here is what is needed as it seems there are important packages missed by the Mint dev team:
> 
> `sudo apt-get install libpango1.0-dev libcairo2-dev`

Thanks it helped but first you need to remove sdrpp with ```sudo apt -fix-broken install``` then run this command, thank you so muchhhhh man

**Reply by gothicserge** [reply]
> Наконец-то мне удалось установить этот пакет. Вот что требуется, так как, похоже, команда разработчиков Mint пропустила некоторые важные пакеты:
> 
> `sudo apt-get install libpango1.0-dev libcairo2-dev`

It didn't help me. It still requires libvolk. What are my other options? I tried downloading and installing the libvolk2 package, but it won't install.

**Reply by Hue-Jhan** [reply]
> > Наконец-то мне удалось установить этот пакет. Вот что требуется, так как, похоже, команда разработчиков Mint пропустила некоторые важные пакеты:
> > `sudo apt-get install libpango1.0-dev libcairo2-dev`
> 
> It didn't help me. It still requires libvolk. What are my other options? I tried downloading and installing the libvolk2 package, but it won't install.

i solved it by installing another version of the same OS, instead of Ubuntu focal i went for jammy if im not wrong, after uninstalling some packages and running some commands shown above.

**Reply by AlexandreRouma** [reply]
This is complete bullshit. The  deb installs all basic  dependencies, you shouldn't be installing drivers for SDRs you might  not  even  have.

**Comment by BareMetalSystems** [comment]
Hi, I am trying to install SDRPP on a RPi5 running the latest Ubuntu desktop build with a MATE desktop. I get the same libvolk2-dev dependency missing error. I have tried the solution here. Have also tried installing libvolk2-dev and libvolk-dev. All without success. Any help would be much appreciated. Many Thanks. Kim

**Reply by BareMetalSystems** [reply]
Sorry - typo - should read - Have also tried installing libvolk2-dev, libvolk1-dev  and libvolk-dev.

**Comment by BareMetalSystems** [comment]
Many Thanks. The project here is on the back burner
      for the moment - more higher priority items ahead of it. I will
      get back on to it early next year and will email you as to the
      results.
      
      73s de Kim
      VK6KL
    
    On 16/12/25 22:01, Hue wrote:
    
    
      
      
        I finally got the package to install. Here is what
          is needed as it seems there are important packages missed by
          the Mint dev team:
        sudo apt-get install
            libpango1.0-dev libcairo2-dev
      
      Thanks it helped but first you need to remove sdrpp
        with sudo apt -fix-broken install
        then run this command, thank you so muchhhhh man
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you commented.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/AlexandreRouma/SDRPlusPlus/discussions/1447#discussioncomment-15269789",
"url": "https://github.com/AlexandreRouma/SDRPlusPlus/discussions/1447#discussioncomment-15269789",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by Hue-Jhan** [comment]
nvm i solved it on Dragon OS focal guys, dont use the sdrpp_ubuntu_focal_amd64.deb but rather the sdrpp_debian_sid_amd64.deb and follow the guy's instructions:
```sudo apt-get install libpango1.0-dev libcairo2-dev```
dont try to even use libvolk

**Reply by gothicserge** [reply]
> Ладно, я решил проблему на Dragon OS Focal, ребята, не используйте sdrpp_ubuntu_focal_amd64.deb, а лучше sdrpp_debian_sid_amd64.deb и следуйте инструкциям этого парня: `sudo apt-get install libpango1.0-dev libcairo2-dev` Даже не пытайтесь использовать libvolk.

This package is installed, but it doesn't work. Other packages require libvolk2

**Comment by AlexandreRouma** [comment]
I wasn't paying attention to this thread because I assumed someone would just give the correct solution without me having to explain basic linux bullshit, but **your distro is based on ubuntu noble, not ubuntu jammy, just google your distro**.
Everyone  here is having the same issue, you are trying to install a deb **for the wrong distro version, which means the library  versions won't be compatible**.
<img width="312" height="107" alt="image" src="https://github.com/user-attachments/assets/665260b1-93e8-4571-ac11-a57bdaa68f3c" />
